compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java

```diff
@@ -321,48 +368,96 @@ private static boolean isCallToAnyStringBuilderMethod(AbstractInsnNode insn) {
                 && ((MethodInsnNode) insn).owner.equals(Type.getInternalName(StringBuilder.class));
     }
 
-    private static AbstractInsnNode convertConstructorCall(TypeInsnNode typeInsn,
-            MethodNode ctorMethod, ClassNode owner, NeoMethod callingNeoMethod,
-            CompilationUnit compUnit) throws IOException {
+    private static AbstractInsnNode convertStructConstructorCall(TypeInsnNode typeInsnNode, MethodNode ctorMethod,
+            ClassNode structClassNode, NeoMethod callingNeoMethod, CompilationUnit compUnit) throws IOException {
 
-        NeoMethod calledNeoMethod;
-        String ctorMethodId = NeoMethod.getMethodId(ctorMethod, owner);
+        SuperNeoMethod calledNeoMethod;
+        String ctorMethodId = NeoMethod.getMethodId(ctorMethod, structClassNode);
+        int fieldSize = structClassNode.fields.size();
         if (compUnit.getNeoModule().hasMethod(ctorMethodId)) {
             // If the module already contains the converted ctor.
-            calledNeoMethod = compUnit.getNeoModule().getMethod(ctorMethodId);
+            calledNeoMethod = (SuperNeoMethod) compUnit.getNeoModule().getMethod(ctorMethodId);
```

Comment:
If the method was already compiled it is retrieved from the NeoModule. That's fine, but then the fieldSize is not set to the correct value if the class is inheriting fields from parent classes.

Reply:
You're absolutely right, I'll add an according test and fix this.